helix-term/src/commands.rs

```diff
@@ -2209,9 +2209,27 @@ fn global_search(cx: &mut Context) {
                         });
                 });
 
+                let user_entered_regex = regex.to_string();
+                let case_sensitive = if smart_case {
+                    user_entered_regex
+                        .chars()
+                        .any(|char| char.is_ascii_uppercase())
+                } else {
+                    true
+                };
                 cx.jobs.callback(async move {
                     let call = move |_: &mut Editor, compositor: &mut Compositor| {
+                        let picker_title = format!(
+                            "Global Search: '{}' {}",
+                            regex,
+                            if case_sensitive {
+                                "[Case Sensitive]"
+                            } else {
+                                "[Case Insensitive]"
+                            }
+                        );
```

Comment:
With a long regex, this will be quite unreadable, the length of the displayed regex should be limited to the N (to determine) first chars of it.